lib/StyleManager.vala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2024 elementary, Inc. (https://elementary.io)
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+
+/**
+ * A class for managing the style of the application. This handles switching light and dark mode based
+ * based on system preference or application preference (see {@link color_scheme_override}), etc.
+ */
+public class Granite.StyleManager : Object {
+    private static Gtk.CssProvider? base_provider = null;
+    private static Gtk.CssProvider? dark_provider = null;
+    private static Gtk.CssProvider? app_provider = null;
+
+    private static GLib.Once<Granite.StyleManager> instance;
+    public static unowned Granite.StyleManager get_instance () {
+        return instance.once (() => {
+            return new Granite.StyleManager ();
+        });
+    }
+
+    /**
+     * If this is set to NO_PREFERENCE the systems preferred color scheme will be used.
+     * Otherwise the color scheme set here will be used.
+     */
+    public Settings.ColorScheme color_scheme_override { get; set; default = NO_PREFERENCE; }
```

Comment:
Maybe just `color_scheme` same as Adw?

Reply:
Tbh I found that a bit ambiguous. Is it the color scheme that currently, actually applies to the application? Is it the override? Is it the system one? It's not that bad but I kinda prefer it more explicitly idk :sweat_smile: 
But we can change it :shrug:

Reply:
Fair enough I updated it :)